app/shared/src/commonMain/kotlin/ui/subject/episode/video/sidesheet/EditDanmakuRegexFilterSideSheet.kt

```diff
@@ -0,0 +1,181 @@
+package me.him188.ani.app.ui.subject.episode.video.sidesheet
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.FlowRow
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.navigationBarsPadding
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.rounded.Close
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.input.key.Key
+import androidx.compose.ui.input.key.KeyEvent
+import androidx.compose.ui.input.key.key
+import androidx.compose.ui.input.key.onKeyEvent
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.unit.dp
+import me.him188.ani.app.data.models.danmaku.DanmakuRegexFilter
+import me.him188.ani.app.ui.subject.episode.TAG_EPISODE_SELECTOR_SHEET
+import me.him188.ani.app.ui.subject.episode.video.settings.DanmakuRegexFilterState
+import me.him188.ani.app.ui.subject.episode.video.settings.EpisodeVideoSettingsSideSheet
+import me.him188.ani.app.ui.subject.episode.video.settings.RegexFilterItem
+import me.him188.ani.utils.platform.Uuid
+
+
+@Composable
+fun EditDanmakuRegexFilterSideSheet(
+    danmakuRegexFilterState: DanmakuRegexFilterState,
+    onDismissRequest: () -> Unit,
+    modifier: Modifier = Modifier,
+    expanded: Boolean = true,  // Use the expanded parameter
+) {
+    val focusManager = LocalFocusManager.current
+    var regexTextFieldValue by rememberSaveable { mutableStateOf("") }
+    var regexTextFieldOutlineTitleText by rememberSaveable { mutableStateOf("填写用于屏蔽的正则表达式，例如：‘.*签.*’ 会屏蔽所有含有文字‘签’的弹幕。") }
+    var isError by rememberSaveable { mutableStateOf(false) }
```

Comment:
```suggestion
    val isError by remember { derivedStateOf { regexTextFieldValue.isBlank() } }
```

一般来说不要使用 `var` 这种 "回调" 的更新模式, 用 val derivedState 是让逻辑明显更清晰的